### ISSUE TITLE
KeyError: 'SSFA is not in the neck registry'

### ISSUE THREAD

**Issue by sunwonHong** [Vegeta2020/SE-SSD#21]
Thank you for sharing the code.

I am trying to train and test.

But i had a problem while training and testing with KITTI.

The issue is like this.
![Screenshot from 2021-07-28 00-21-31](https://user-images.githubusercontent.com/80449955/127181735-455a52f6-aadd-4e49-af43-be2fd475dc75.png)

What is wrong with me?

If anyone knows what error it is, please let me know how to solve it.

Thank you.

**Comment by Vegeta2020** [comment]
SSFA module is in SE-SSD/det3d/models/necks/rpn_v1.py. Please ensure the installation with setup.py or citation in SE-SSD/det3d/models/necks/__init__.py.

**Comment by sunwonHong** [comment]
Ok. Let me try again.

Thank you for replying.

**Comment by sunwonHong** [comment]
@Vegeta2020 

It was solved when i did setup.py again thanks.

But i got a different error.
Do you or anyone know what problem it is?

![Screenshot from 2021-08-05 19-20-07](https://user-images.githubusercontent.com/80449955/128334795-88b00161-1843-41aa-9775-2b0515905bef.png)

... skip ...

![Screenshot from 2021-08-05 19-20-35](https://user-images.githubusercontent.com/80449955/128334801-1d060f9b-1c35-47f4-b2e7-be6fbff78cb6.png)

... skip ...

![Screenshot from 2021-08-05 19-20-53](https://user-images.githubusercontent.com/80449955/128334804-7551cc75-0502-40ac-8019-be2292ce8171.png)

... skip ...

![Screenshot from 2021-08-05 19-22-13](https://user-images.githubusercontent.com/80449955/128334807-075c487b-b64b-4509-87bc-19758da33608.png)
![Screenshot from 2021-08-05 19-22-32](https://user-images.githubusercontent.com/80449955/128334811-982f1d14-d025-40c0-a482-e241ca52d7d3.png)

What is this problem from?
TypeError: can't pickle _thread.RLock objects

Please let me know if you know what it is.

Thank you.

**Comment by sunwonHong** [comment]
It was solved.
#14 helped me.

**Comment by caroltk** [comment]
Hello, I also met this problem. 
I am wondering which setup.py did you run to solve the KeyError: 'SSFA is not in the neck registry'?

$ cd ./SE-SSD/det3d/core/iou3d
$ python setup.py install
or
$ cd ./SE-SSD
$ python setup.py build develop

Thank you so much for the help.

**Comment by sunwonHong** [comment]
I did both.
Make sure to do it in order.
I think the order is important to run.

**Comment by caroltk** [comment]
Thank you for the quick reply. It was solved. 
And I am also wondering about your PyTorch version?

Best Regards.

**Comment by sunwonHong** [comment]
I am using 1.7.1.
